### DISCUSSION TITLE
Poll items for all users

### DISCUSSION THREAD

**Discussion by carlosvirreira** [Shelf-nu/shelf.nu#1328] [Ideas]
We want to poll our userbase on what things the team should focus on, work next: 

My top 5 (with no particular order)
- Signed Custody Release (Prior to assigning custody, attach an agreement, the user must sign this agreement, the sign certificate is stored in shelf)
- Asset Alerts (Separate area of shelf, where you can set time based, action based or tag based alerts to get triggered on shelf and outside of shelf)
- Spot Checks (Select a list of items and initiate a check - the timer runs until all assets have had its tags scanned - shelf generates a PDF that can be used as proof of the tags being scanned)
- Alternate ID for asset (Allows you to source or take over QR codes)

**Comment by Joedon3** [comment]
- Add asset images from mobile camera
- Asset Alerts (asset changed, asset expiration, EOL dates.  QR scan alert)
- 3rd party integration (start with webhooks)
           - This could be the start of Asset Alerts.
- Report generation, export to pdf
- Read-only users, role based by category

**Reply by carlosvirreira** [reply]
> * Add asset images from mobile camera
**On mobile you can click on the asset image / then you have an (edit image) button that will lead you to the edit asset page where you can then click on "Photo library / Take photo / Choose file" - do I interpret correctly that you would prefer to avoid that middle step and add a main image faster? Or do you want to add _multiple_ images per asset?**

> * Asset Alerts (asset changed, asset expiration, EOL dates.  QR scan alert)
**We have some concepts for this Joe. Do you prefer to set the alert 'asset per asset' or have a dedicated area to configure and map assets that share the same alerts?**

> * 3rd party integration (start with webhooks)
>   - This could be the start of Asset Alerts.


> * Report generation, export to pdf

> * Read-only users, role based by category


Thanks for your input!

**Reply by Joedon3** [reply]
> > * Add asset images from mobile camera
> >   **On mobile you can click on the asset image / then you have an (edit image) button that will lead you to the edit asset page where you can then click on "Photo library / Take photo / Choose file" - do I interpret correctly that you would prefer to avoid that middle step and add a main image faster? Or do you want to add _multiple_ images per asset?**
> >   -  _It just might be my device, but I only have a choose file button.  My picker doesn't list my camera.   Multiple images would be useful as well, but there are already a couple feature requests related to that.  My input on that is to allow documents, not just images. :)_
> 
> > * Asset Alerts (asset changed, asset expiration, EOL dates.  QR scan alert)
> >   **We have some concepts for this Joe. Do you prefer to set the alert 'asset per asset' or have a dedicated area to configure and map assets that share the same alerts?**
> >   - _Asset by asset could become very time consuming and difficult to manage.  I think a separate 'Alerts' page would make things easier to implement in the long run, and easier to update.  The more I think about it, the more I see the end result similar to a SIEM dashboard where you can run some type of generic query and have a report generated, or an immediate alert created.  Maybe something like that is in the next major release.  :)_
> 
> Thanks for your input!

Thanks for an awesome product and community involvement!!!

**Comment by DonKoko** [comment]
My 2 cents on your suggestions @carlosvirreira :

>     Signed Custody Release (Prior to assigning custody, attach an agreement, the user must sign this agreement, the sign certificate is stored in shelf)
I think its about time we finish this. Its like 90% done already. 

>     Asset Alerts (Separate area of shelf, where you can set time based, action based or tag based alerts to get triggered on shelf and outside of shelf)
This is cool but I think it needs a lot of definition. Also we currently only send "notifications" via email. Doing something like this I think will be good to be combined with notifications

>     Spot Checks (Select a list of items and initiate a check - the timer runs until all assets have had its tags scanned - shelf generates a PDF that can be used as proof of the tags being scanned)
I don't get the purpose of the timer but overall this is a quite simple feature because we already have the bulk scanner.

>     Alternate ID for asset (Allows you to source or take over QR codes)
I would avoid this like the plague and rather prioritize adding barcodes or simple/internal qr, 2D matrix or sth like this. I think I need to think further about how we could combine both things somehow. Also important thing, our DB is designed to support multiple codes per asset so we don't have to ever do stuff like "taking over" the default shelf QR code.

**Reply by carlosvirreira** [reply]
My replies below: 

> My 2 cents on your suggestions @carlosvirreira :
> 
> > ```
> > Signed Custody Release (Prior to assigning custody, attach an agreement, the user must sign this agreement, the sign certificate is stored in shelf)
> > ```
> 
> I think its about time we finish this. Its like 90% done already.

**Sounds good. Let's then please review together where we ended up.**

> 
> > ```
> > Asset Alerts (Separate area of shelf, where you can set time based, action based or tag based alerts to get triggered on shelf and outside of shelf)
> > ```
> 
> This is cool but I think it needs a lot of definition. Also we currently only send "notifications" via email. Doing something like this I think will be good to be combined with notifications
> 

**Notifications, mails, sms - all are suitable. I was thinking that to start, a mail alert would suffice. It has a lot of impact. However, if we can render that alert in an app notification area it would be helpful for overviewing.**

> > ```
> > Spot Checks (Select a list of items and initiate a check - the timer runs until all assets have had its tags scanned - shelf generates a PDF that can be used as proof of the tags being scanned)
> > ```
> 
> I don't get the purpose of the timer but overall this is a quite simple feature because we already have the bulk scanner.

**Timer can honestly be skipped on front end. I think that if we generate a PDF report or some sort of proof of completion, that the time it took to audit could be an interesting metric.**

> 
> > ```
> > Alternate ID for asset (Allows you to source or take over QR codes)
> > ```
> 
> I would avoid this like the plague and rather prioritize adding barcodes or simple/internal qr, 2D matrix or sth like this. I think I need to think further about how we could combine both things somehow. Also important thing, our DB is designed to support multiple codes per asset so we don't have to ever do stuff like "taking over" the default shelf QR code.

**Alright - in my eyes if you have an existing QR code on an asset (or barcode // datamatrix) you could note down the string of numbers/characters (no url) and add it as an alternate ID then if the in app scanner gets open and we find that ID we can direct to the right asset page**

**Comment by carlosvirreira** [comment]
Had a demo with a manufacturer today. Scheduling alerts for individual assets gets a + 1 from them. (Calibrations, etc).

**Comment by carlosvirreira** [comment]
One feature that would be really useful is the ability to sync with Google Calendar. T
